pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -75,10 +74,7 @@ func (ctrl *csiSnapshotSideCarController) syncContent(content *crdv1.VolumeSnaps
 	} else {
 		if content.Spec.Source.VolumeHandle != nil && content.Status == nil {
 			klog.V(5).Infof("syncContent: Call CreateSnapshot for content %s", content.Name)
-			if err = ctrl.createSnapshot(content); err != nil {
-				ctrl.updateContentErrorStatusWithEvent(content, v1.EventTypeWarning, "SnapshotCreationFailed", fmt.Sprintf("Failed to create snapshot with error %v", err))
-				return err
-			}
+			ctrl.createSnapshot(content)
```

Comment:
These functions will never return an error because they just schedule async operations. Each async operation creates an event upon any failures. Deletion seems to be implemented similarly, except it always just passes the nil error further up the stack.